plugins/action/dcnm_network.py

```diff
@@ -0,0 +1,586 @@
+
+    MSD Configuration Features:
+    - Automatic deploy flag inheritance from parent to child (no override allowed)
+    - VRF and L2-only settings propagation to child fabrics
+    - State transformation (parent 'overridden' becomes child 'replaced')
+    - Attachment restriction enforcement (parent-only configuration)
+    - Deploy flag restriction enforcement (parent-only configuration)
+    - Comprehensive validation of fabric hierarchy relationships
+
+    Error Handling:
+    - Fail-fast execution with detailed error messages
+    - Comprehensive validation before any execution begins
+    - Structured error reporting with fabric context information
+    """
+
+    def run(self, tmp=None, task_vars=None):
+        """
+        Main entry point for the action plugin.
+
+        This method orchestrates the MSD fabric detection and processing workflow:
+        1. Validates required parameters
+        2. Detects fabric type using NDFC fabric associations API
+        3. Validates and splits MSD parent/child configurations
+        4. Executes network operations on parent and child fabrics
+
+        Args:
+            tmp (str, optional): Temporary directory path for file operations
+            task_vars (dict, optional): Ansible task variables and context
+
+        Returns:
+            dict: Ansible result dictionary containing:
+                - changed (bool): Whether any changes were made
+                - failed (bool): Whether the operation failed
+                - msg (str): Error message if failed
+                - workflow (str): Type of workflow executed
+                - parent_fabric (dict): Parent fabric results for MSD
+                - child_fabrics (list): Child fabric results for MSD
+        """
+        if task_vars is None:
+            task_vars = dict()
+
+        result = dict(
+            changed=False
+        )
+
+        # Get module arguments
+        module_args = self._task.args.copy()
+        fabric_name = module_args.get('fabric')
+
+        if not fabric_name:
+            result['failed'] = True
+            result['msg'] = "fabric parameter is required"
+            return result
+
+        # Detect fabric type using fabric associations API
+        fabrics = self._get_fabric_associations(task_vars)
+
+        if fabrics is None:
+            result['failed'] = True
+            result['msg'] = "Failed to get fabric associations"
+            return result
+
+        # Validate fabric hierarchy before processing
+
+        state = module_args.get('state')  # Get the state parameter
+        if not state:
+            result['failed'] = True
+            result['msg'] = "The 'state' parameter is required"
+            return result
+
+        config = module_args.get('config')
+
+        if not isinstance(config, list) or not config:
+            # For 'query' and 'deleted', allow empty config (interpreted as all networks)
+            if state in ['query', 'deleted']:
```

Comment:
We need to support `[]` for state `overridden` also. When we pass an empty list it should remove all networks